### PR TITLE
[msg] Update message version and session id fields to match spec.

### PR DESCRIPTION
**Pull request by turon**
#### Problem
The message format does not fully match what is described in the specification.

#### Change overview
* Update message version field from `2` to `0` to match spec.
* Rename KeyId to SessionId
* Move SessionId field up to fixed portion of header
* Rename EncryptionType to SessionType
* Remapping of auth tag length based on SessionId rather than EncryptionType deferred to follow-on.

#### Testing
Standard CI unit test suite.
Ran echo-request / responder locally.

### CODE REVIEW

File: src/protocols/secure_channel/PASESession.h [hunk 1]
```diff
@@ -340,13 +340,13 @@ class SecurePairingUsingTestSecret : public PairingSession
     SecurePairingUsingTestSecret()
     {
         SetLocalKeyId(0);
```
Comment (kghost) [inline src/protocols/secure_channel/PASESession.h:342]: Shouldn't we also rename this one to `SessionId` ?

Anyway, It is fine here, we can do it in following PRs

**Review by kghost — APPROVED** [review]
This is great, which we should have done long ago.

File: src/protocols/secure_channel/PASESession.h (hunk 1 above)
Comment (turon) [reply src/protocols/secure_channel/PASESession.h]: Updated.

**Review by andy31415 — APPROVED** [review]

File: src/transport/raw/MessageHeader.h [hunk 2]
```diff
@@ -292,7 +292,7 @@ class PacketHeader
 
 private:
     /// Represents the current encode/decode header version
-    static constexpr int kHeaderVersion = 2;
+    static constexpr int kMsgHeaderVersion = 0;
```
Comment (turon) [inline src/transport/raw/MessageHeader.h:295]: Primary change #1.  (To elevate above rename noise).

File: src/transport/raw/MessageHeader.cpp [hunk 3]
```diff
@@ -151,10 +151,13 @@ CHIP_ERROR PacketHeader::Decode(const uint8_t * const data, uint16_t size, uint1
...
     version = ((header & kVersionMask) >> kVersionShift);
-    VerifyOrExit(version == kHeaderVersion, err = CHIP_ERROR_VERSION_MISMATCH);
+    VerifyOrExit(version == kMsgHeaderVersion, err = CHIP_ERROR_VERSION_MISMATCH);
 
     mFlags.SetRaw(header);
-    mEncryptionType = static_cast<Header::EncryptionType>((header & kEncryptionTypeMask) >> kEncryptionTypeShift);
+    mSessionType = static_cast<Header::SessionType>((header & kSessionTypeMask) >> kSessionTypeShift);
+
+    err = reader.Read16(&mSessionId).StatusCode();
+    SuccessOrExit(err);
```
Comment (turon) [inline src/transport/raw/MessageHeader.cpp:160]: Primary change #2a: move Session ID field to before Message Counter in Decode.

File: src/transport/raw/MessageHeader.cpp [hunk 4]
```diff
@@ -271,11 +271,12 @@ CHIP_ERROR PacketHeader::Encode(uint8_t * data, uint16_t size, uint16_t * encode
...
         .Set(Header::FlagValues::kDestinationNodeIdPresent, mDestinationNodeId.HasValue());
 
-    uint16_t header = (kHeaderVersion << kVersionShift) | encodeFlags.Raw();
-    header |= (static_cast<uint16_t>(static_cast<uint16_t>(mEncryptionType) << kEncryptionTypeShift) & kEncryptionTypeMask);
+    uint16_t header = (kMsgHeaderVersion << kVersionShift) | encodeFlags.Raw();
+    header |= (static_cast<uint16_t>(static_cast<uint16_t>(mSessionType) << kSessionTypeShift) & kSessionTypeMask);
 
     uint8_t * p = data;
     LittleEndian::Write16(p, header);
+    LittleEndian::Write16(p, mSessionId);
```
Comment (turon) [inline src/transport/raw/MessageHeader.cpp:279]: Primary change #2b: move Session ID field to before Message Counter in Encode.

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @msandstedt ?

**Review by msandstedt — APPROVED** [review]

**Pull request merged**